uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/controllers/sub-controllers/leave-request-ctrls.spec.js

```diff
@@ -1288,6 +1288,137 @@
         });
       });
 
+      describe('getStatuses', function () {
+        var status = {};
+
+        beforeEach(function () {
+          var collectionName = 'hrleaveandabsences_leave_request_status';
+          var allStatuses = optionGroupMock.getCollection(collectionName);
+
+          status = _.indexBy(allStatuses, 'name');
+
+          initTestController({
+            contactId: CRM.vars.leaveAndAbsences.contactId,
+            userRole: 'admin'
+          });
+
+          $ctrl._init();
+          $ctrl.request = {
+            status_id: null
+          };
+          $ctrl.requestStatuses = status;
+        });
+
+        it('defines a getStatuses method', function () {
+          expect(typeof $ctrl.getStatuses).toBe('function');
+        });
+
+        it('returns an empty array if .request is not defined', function () {
+          delete $ctrl.request;
```

Comment:
The following structure should be followed
```javascript
 describe('when request is not defined', function() {
 	beforeEach(function(){
 	  delete $ctrl.request;
 	  //Or any other setup
 	});

 	it('returns an empty array', function () {
   // it block should contain only the expect sentence, or any other statement, which cannot be moved to beforeEach or AfterEach
	  var result = $ctrl.getStatuses();
	  expect(result).toEqual([]);
	  //or directly, I would prefer this approach
	  expect($ctrl.getStatuses()).toEqual([]);
	});
 })
```

Please apply this structure for all your unit tests.